### PR TITLE
Document the two histories of domains

### PR DESCRIPTION
**Pull request by PaulWessel**
Without getting too technical, there is a different treatment for plot region and data regions.  See if this makes sense.  I put it under the command line history section.

### CODE REVIEW

File: doc/rst/source/cookbook/features.rst [hunk 1]
```diff
@@ -440,7 +440,13 @@ do not need to repeat the region and projection information, as shown here::
      gmt end show
 
 Thus, the chosen options remain in effect until you provide new option
-arguments on the command line.
+arguments on the command line.  **Note**: We keep track of two types of regions,
+One is the domain used for a map and one is the domain used for a grid.
```
Comment (meghanrjones) [inline doc/rst/source/cookbook/features.rst:444]:
```suggestion
One is the domain used for a map and one is the domain used for a grid,
```

File: doc/rst/source/cookbook/features.rst [hunk 2]
```diff
@@ -440,7 +440,13 @@ do not need to repeat the region and projection information, as shown here::
...
+which often are the same.  When a plot is specified without providing
+a region then we look for a previous plot region in the history first, and
+if it is not found then we look for a grid domain to use instead.  However,
```
Comment (meghanrjones) [inline doc/rst/source/cookbook/features.rst:447]: So based on the grid domain description, if a plot region is not found a table region will not work (e.g., blockmean)?

Comment (PaulWessel) [reply doc/rst/source/cookbook/features.rst:447]: What I mean to say is that RG will be used if there is no RP, but RP will never be used if there is no RG.

File: doc/rst/source/cookbook/features.rst [hunk 3]
```diff
@@ -440,7 +440,13 @@ do not need to repeat the region and projection information, as shown here::
      gmt end show
 
 Thus, the chosen options remain in effect until you provide new option
-arguments on the command line.
+arguments on the command line.  **Note**: We keep track of two types of regions,
+One is the domain used for a map and one is the domain used for a grid,
```
Comment (meghanrjones) [inline doc/rst/source/cookbook/features.rst:444]:
```suggestion
One is the domain used for a map and one is the domain used for processing,
```

File: doc/rst/source/cookbook/features.rst [hunk 4]
```diff
@@ -440,7 +440,13 @@ do not need to repeat the region and projection information, as shown here::
...
+which often are the same.  When a plot is specified without providing
+a region then we look for a previous plot region in the history first, and
+if it is not found then we look for a grid domain to use instead.  However,
```
Comment (meghanrjones) [inline doc/rst/source/cookbook/features.rst:447]:
```suggestion
if it is not found then we look for the processing domain to use instead.  However,
```

File: doc/rst/source/cookbook/features.rst [hunk 5]
```diff
@@ -440,7 +440,13 @@ do not need to repeat the region and projection information, as shown here::
...
+if a data-processing module is not given a region then we only look
+for a previous data domain; we never substitute a plot domain in that case.
```
Comment (meghanrjones) [inline doc/rst/source/cookbook/features.rst:449]:
```suggestion
for a previous processing domain; we never substitute a plot domain in that case.
```

**Review by meghanrjones — APPROVED** [review]
A couple suggestions based on my interpretation that RG can be set in the history regardless of whether the region was set on a table or grid.

**Pull request merged**